### ISSUE TITLE
workflow.py:2080 ERROR    'favorites'

### ISSUE THREAD

**Issue by politicus** [tomy0000000/coon#1]
I tried the workflow and couldn't make it work. I couldn't say whether it is because of a bug of because I missed a configuration step. 

I imported the workflow added the APP_ID variable before running the workflow, didn't add any currency to the favorites list as I couldn't. I didn't work. 

Apparently there is a problem with the favorites as I got this error. 

```
[23:58:14.285] Productivity Coon[Script Filter] Queuing argument '(null)'
[23:58:18.468] Productivity Coon[Script Filter] Script with argv '(null)' finished
[23:58:18.474] STDERR: Productivity Coon[Script Filter] .
23:58:17 workflow.py:2061 DEBUG    ---------- Productivity Coon (1.0.0) ----------
23:58:17 workflow.py:1468 DEBUG    reading settings from /Users/Username/Library/Application Support/Alfred/Workflow Data/tech.tomy.coon/settings.json
23:58:17 workflow.py:2338 INFO     checking for update ...
23:58:17 background.py:233 DEBUG    [__workflow_update_check] command cached: /Users/Username/Library/Caches/com.runningwithcrayons.Alfred/Workflow Data/tech.tomy.coon/__workflow_update_check.argcache
23:58:17 background.py:237 DEBUG    [__workflow_update_check] passing job to background runner: [u'/usr/bin/python', '/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/workflow/background.pyc', u'__workflow_update_check']
.
23:58:17 workflow.py:2061 DEBUG    ---------- Productivity Coon (1.0.0) ----------
23:58:17 background.py:243 DEBUG    [__workflow_update_check] background job started
.
23:58:17 workflow.py:2061 DEBUG    ---------- Productivity Coon (1.0.0) ----------
23:58:17 workflow.py:2338 INFO     checking for update ...
23:58:17 background.py:225 INFO     [__workflow_update_check] job already running
23:58:18 query.py:72 INFO     Run Pattern 0
23:58:18 workflow.py:2080 ERROR    'favorites'
Traceback (most recent call last):
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/workflow/workflow.py", line 2073, in run
    func(self)
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/__init__.py", line 66, in convert
    query.run_pattern(workflow, rates)
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/query.py", line 74, in run_pattern
    func(workflow, rates)
  File "/Users/```
[23:58:14.285] Productivity Coon[Script Filter] Queuing argument '(null)'
[23:58:18.468] Productivity Coon[Script Filter] Script with argv '(null)' finished
[23:58:18.474] STDERR: Productivity Coon[Script Filter] .
23:58:17 workflow.py:2061 DEBUG    ---------- Productivity Coon (1.0.0) ----------
23:58:17 workflow.py:1468 DEBUG    reading settings from /Users/Username/Library/Application Support/Alfred/Workflow Data/tech.tomy.coon/settings.json
23:58:17 workflow.py:2338 INFO     checking for update ...
23:58:17 background.py:233 DEBUG    [__workflow_update_check] command cached: /Users/Username/Library/Caches/com.runningwithcrayons.Alfred/Workflow Data/tech.tomy.coon/__workflow_update_check.argcache
23:58:17 background.py:237 DEBUG    [__workflow_update_check] passing job to background runner: [u'/usr/bin/python', '/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/workflow/background.pyc', u'__workflow_update_check']
.
23:58:17 workflow.py:2061 DEBUG    ---------- Productivity Coon (1.0.0) ----------
23:58:17 background.py:243 DEBUG    [__workflow_update_check] background job started
.
23:58:17 workflow.py:2061 DEBUG    ---------- Productivity Coon (1.0.0) ----------
23:58:17 workflow.py:2338 INFO     checking for update ...
23:58:17 background.py:225 INFO     [__workflow_update_check] job already running
23:58:18 query.py:72 INFO     Run Pattern 0
23:58:18 workflow.py:2080 ERROR    'favorites'
Traceback (most recent call last):
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/workflow/workflow.py", line 2073, in run
    func(self)
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/__init__.py", line 66, in convert
    query.run_pattern(workflow, rates)
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/query.py", line 74, in run_pattern
    func(workflow, rates)
  File "/Users/Username/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/query.py", line 82, in pattern_0
    for currency in settings["favorites"]:
KeyError: 'favorites'
23:58:18 workflow.py:2082 INFO     for assistance, see: https://git.io/fjD6M
23:58:18 workflow.py:2103 DEBUG    ---------- finished in 0.840s ----------
23:58:18 workflow.py:2254 DEBUG    set last run version: 1.0.0
23:58:18 workflow.py:2103 DEBUG    ---------- finished in 1.464s ----------
[23:58:18.495] Productivity Coon[Script Filter] {"items": [{"valid": false, "subtitle": "'favorites'", "icon": {"path": "/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"}, "title": "Error in workflow 'Productivity Coon'"}]}

```/Dropbox/Applications_Backups/Alfred_Backups/Backup_Alfred/Alfred.alfredpreferences/workflows/user.workflow.08009832-5D2F-4E5A-BE10-9B257CBB7595/currency/query.py", line 82, in pattern_0
    for currency in settings["favorites"]:
KeyError: 'favorites'
23:58:18 workflow.py:2082 INFO     for assistance, see: https://git.io/fjD6M
23:58:18 workflow.py:2103 DEBUG    ---------- finished in 0.840s ----------
23:58:18 workflow.py:2254 DEBUG    set last run version: 1.0.0
23:58:18 workflow.py:2103 DEBUG    ---------- finished in 1.464s ----------
[23:58:18.495] Productivity Coon[Script Filter] {"items": [{"valid": false, "subtitle": "'favorites'", "icon": {"path": "/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"}, "title": "Error in workflow 'Productivity Coon'"}]}

```

**Comment by tomy0000000** [comment]
Sorry, that was my bad.

This issue occurred when new install doesn't have an settings,
which I just patched in v1.0.1.

Please try installing the newest release,
and let me know if there's any other problem.

Thanks. 😛

**Comment by politicus** [comment]
Installed the newest version. Working.
